code/game/objects/items/devices/scanners.dm

```diff
@@ -867,3 +867,109 @@ REAGENT SCANNER
 		dat += "<font color='red'>Retinal misalignment detected.</font><BR>"
 
 	return dat
+
+// Randomly polls tiles on the Z level to give a vague indication if whether something is about.
+/obj/item/space_scanner
+	name = "gravimetric analyzer"
+	icon = 'icons/obj/device.dmi'
+	icon_state = "forensic2"
+	desc = "A handheld suite of sensors designed to scan local areas of space for gravimetric signatures. The compact design renders it somewhat unreliable."
+	slot_flags = SLOT_BELT
+	throwforce = 3
+	w_class = WEIGHT_CLASS_SMALL
+	throw_speed = 5
+	throw_range = 10
+	origin_tech = "materials=4;programming=4;magnets=4"
+	var/obj/item/stock_parts/scanning_module/s_module
+	var/next_use = 0
+	var/scan_cd = 10 SECONDS //how long before we can scan again
+	var/scan_tile_count // The number of tiles to poll randomly.
+
+/obj/item/space_scanner/Initialize()
+	s_module = new(null)
+	scan_tile_count = 20 + (s_module.rating * 20)
+
+/obj/item/space_scanner/screwdriver_act(mob/living/user, obj/item/I)
+	to_chat(user, "<span class='notice'> You unscrew and remove the [s_module.name] from [src].</span>")
+	s_module.loc = get_turf(src.loc)
+	s_module = null
+	return
+
+/obj/item/space_scanner/attackby(obj/item/W, mob/user, params)
+	if(istype(W, /obj/item/stock_parts/scanning_module))
+		if(!s_module)
+			user.drop_item()
+			W.loc = src
+			s_module = W
+			to_chat(user, "<span class='notice'>You install a [s_module.name] in [src].</span>")
+			scan_tile_count = 20 + (s_module.rating * 20)
```

Comment:
Make this into a proc. It's being used twice now already so it's best to make a proc for this. Avoids mistakes etc